### ISSUE TITLE
Allow plugin for custom logging ....

### ISSUE THREAD

**Issue by ChristianVirtual** [FoldingAtHome/fah-issues#1450]
The current logging files are very ... rich in content. Parsing those for 3rd party apps is quite some task. It would be easier and faster for 3rd party API user if an alternative more technically logging format could be established. Also in parallel to the current one for those using the FAHControl .

Also a remote command to rotate the log file could help and when a 3rd paty detect the file get too big it could request to cycle into a new logfile like in case of client restart.

**Comment by PantherX** [comment]
I guess the first part about parsing log could be taken care in #1448 

What's your definition of big? When I last tested, the log file was 100 MBs and FAHControl just about caved in but it survived. I do like the idea and maybe it can be implemented in both, a button in FAHControl "Create New Log" and an API call.

**Comment by ChristianVirtual** [comment]
#1272 come close. I',m talking about log file with several MB (> 5MB). When a system is technically stable and crunch day-in/day-out this goes quickly. Every-time I open my 3rd party App I need to transfer quite some amount of data and reprocess without adding much value. 

Ways to optimize: 
1) compress the log file before sending via API (but not that easy via Telnet; this compressing might be also useful for upping viewer as trajectories are also very large
2) provide the logged content is a more machine-friendly format (result in less bytes and faster processing)

**Comment by shorttack** [comment]
@ChristianVirtual 1. You can get troubleshooting and usage assistance at https://foldingforum.org/index.php.
2. Have you played with the Verbosity command under the Expert tab? Reducing log verbosity in a stable system will reduce the megabytes you have to crunch daily.

**Comment by ChristianVirtual** [comment]
Normally I leave the verbosity on default (3?). It’s not so much about troubleshooting on concrete cases. It’s to have the chance to increase the user experience for 3rd party apps (like HFM or my FAHMM for iOS and soon android). As a mobile addition to FAHControl. 
I need to reprocess the log file periodically to catch up with finished WU an der update local databases on devices. This reprocessing and continued writing into those databases takes time as the log format is right now more intended for humans as required by FAHControl. Was looking for a less wasteful representation for more automatic processing. 
When loading the current log file via 3rd party API I have only the option to start listening what’s going on right now or refresh from begin of current log.
It would be also useful instead if an alternative log format to have an 3rd API to show from certain time stamp with current log. All to reduce transferred bytes to a mobile device and to reduce processing time.

**Comment by shorttack** [comment]
> 3rd party apps (like HFM or my FAHMM for iOS and soon android). As a mobile addition to FAHControl.

Yes, I can see why keeping at Verbosity=3 prevents breaking HFM, etc.

Now I know what you wish: an enhancement to Stats.
